pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -2647,3 +2663,167 @@ func handleDNSDelete(ctxArg interface{}, key string,
 	maybeHandleDNS(ctx)
 	log.Infof("handleDNSDelete done for %s\n", key)
 }
+
+func validateAppNetworkConfig(ctx *zedrouterContext, appNetConfig types.AppNetworkConfig,
+	appNetStatus *types.AppNetworkStatus) bool {
+	log.Infof("AppNetwork(%s), check for duplicate port map acls", appNetConfig.DisplayName)
+	// For App Networks, check for common port map rules
+	ulCfgList0 := appNetConfig.UnderlayNetworkList
+	if len(ulCfgList0) == 0 {
+		return false
+	}
+	sub := ctx.subAppNetworkConfig
+	items := sub.GetAll()
+	for _, cfg := range items {
+		appNetConfig1 := cast.CastAppNetworkConfig(cfg)
+		ulCfgList1 := appNetConfig1.UnderlayNetworkList
+		if appNetConfig.DisplayName == appNetConfig1.DisplayName ||
+			len(ulCfgList1) == 0 {
+			continue
+		}
+		ret, err := checkUnderlayNetworkForDuplicatePortMap(ctx, ulCfgList0,
+			ulCfgList1)
+		if err != nil {
+			addError(ctx, appNetStatus, "portmap for underlay", err)
+			return true
+		}
+		if ret {
+			log.Errorf("app Network(%s) have overlapping portmap rule with %s\n",
+				appNetStatus.DisplayName, appNetConfig1.DisplayName)
+			errStr := fmt.Sprintf("portmap rule overlap with %s",
+				appNetConfig1.DisplayName)
+			err := errors.New(errStr)
+			addError(ctx, appNetStatus, "portmap for underlay", err)
+			return true
+		}
+	}
+	return false
+}
+
+func checkUnderlayNetworkForDuplicatePortMap(ctx *zedrouterContext,
```

Comment:
Please add a comment specifying the return value (bool to indicate if there is a duplicate, and error for any other errors).